### PR TITLE
CRM-19885 / dev/core/issues/36 fix fatal error on scheduled reminders due to no default

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Fix fatal error causing fatal error

Before
----------------------------------------
Fatal error per https://lab.civicrm.org/dev/core/issues/36#note_3509

After
----------------------------------------
No fatal error - note the logger / tester refers to there still being an error - but it appears to be a separate issue & is not fleshed out. per David Tarrant comment here https://issues.civicrm.org/jira/browse/CRM-19885?focusedCommentId=109335&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-109335 this fatal is blocking the scheduled job entirely so even if this just mitigates it is seems better

Technical Details
----------------------------------------
Previous PR by @twomice & discussion #9686 

The mysql commands have been tested & improve the situation per comments on https://lab.civicrm.org/dev/core/issues/36#note_3509

Comments
----------------------------------------
Is this correct for the upgrade commands in new system @totten ?

### CODE REVIEW

**Comment by monishdeb** [comment]
@eileenmcnaughton the reason why there is a test failure is because you need to run regen.sh and include the civicrm_generated.mysql changes in the PR.

**Comment by eileenmcnaughton** [comment]
@monishdeb there were style warnings (now fixed) but rerunning regen didn't create any changes - any ideas (@seamuslee001 )

File: CRM/Upgrade/Incremental/php/FiveTwo.php [hunk 1]
```diff
@@ -67,18 +67,14 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
...
-  //    // Additional tasks here...
-  //    // Note: do not use ts() in the addTask description because it adds unnecessary strings to transifex.
-  //    // The above is an exception because 'Upgrade DB to %1: SQL' is generic & reusable.
-  //  }
+  /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_5_2_0($rev) {
```
Comment (monishdeb) [inline CRM/Upgrade/Incremental/php/FiveTwo.php]: isn't this ``` upgrade_5_2_alpha1($rev)``` ?

Comment (eileenmcnaughton) [reply CRM/Upgrade/Incremental/php/FiveTwo.php]: hmm not too sure @totten ? Also running codegen I don't seem to get the DAO file updates (& sql) I expect

**Comment by seamuslee001** [comment]
@eileenmcnaughton if you can revert the changes to FiveTwo.php then i am happy to merge, the reason is we don't need a function in FiveTwo to have it run the .mysql.tpl file and the .mysql.tpl file is for the Alpha1 version. 

Also reason why no updated DAO is that to assign the default key to the DAO array it does an isset and isset-> on a field with a value of 0 = false and no generated action schedules so we don't need updated sql file

**Comment by eileenmcnaughton** [comment]
pushed - go jenkins go

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by eileenmcnaughton** [comment]
@seamuslee001 tests passing now!

**Comment by seamuslee001** [comment]
Changes are sensible and look good merging

**Pull request merged**